### PR TITLE
Devel/add default gw libmnl

### PR DESCRIPTION
**Pull request by metelik**
Re-re-implementing the add_default_gateway to use libmnl so it would be easier for the run-from-ye-desktop-users

### CODE REVIEW

**Review by adrigonzo — APPROVED** [review]

File: src/netif.c [hunk 1]
```diff
@@ -1724,42 +1724,76 @@ static int remove_all_gateways6(struct netif *nb, const char *ifname)
...
+  rtm->rtm_type     = RTN_UNICAST;
+  rtm->rtm_flags    = RTNH_F_ONLINK;
+
+  {
+    struct sockaddr_in addr;
+    memset(&addr, 0, sizeof(struct sockaddr_in));
+    addr.sin_family = AF_INET;
+
+    if (inet_pton(AF_INET, gateway_ip, &addr.sin_addr) <= 0) {
+      debug("Bad IP address for the default gateway: %s", gateway_ip);
```
Comment (PeterJeaiem) [inline src/netif.c:1753]: want to keep debug ?

File: src/netif.c [hunk 2]
```diff
@@ -1724,42 +1724,76 @@ static int remove_all_gateways6(struct netif *nb, const char *ifname)
...
-    addr->sin_family = AF_INET;
-    addr->sin_addr.s_addr = INADDR_ANY;
+  if (ifname) {
+    int idx = ifname_to_index(nb, ifname);
 
-    route.rt_dev = (char *) ifname;
-    route.rt_flags = RTF_UP | RTF_GATEWAY;
-    route.rt_metric = 0;
+    if (!idx) {
+      debug("[%s %d %s]: No such device: '%s'\r\n", __FILE__, __LINE__, __FUNCTION__, ifname);
```
Comment (PeterJeaiem) [inline src/netif.c:1765]: want to keep debug print ?

**Review by PeterJeaiem — APPROVED** [review]

**Review by alanj853 — APPROVED** [review]

File: src/netif.c (hunk 1 above)
Comment (metelik) [reply src/netif.c:1753]: It is compiled out via the #debug macro

File: src/netif.c (hunk 2 above)
Comment (metelik) [reply src/netif.c:1765]: yup - it is compiled out via the ansi-c #debug pre-compiler's macro ;)

**Pull request merged**